token-metadata/program/src/processor/delegate/delegate.rs

```diff
@@ -406,23 +406,37 @@ fn create_persistent_delegate_v1(
     )?;
 
     // For Utility Delegates we request Close Authority as well so that the
-    // token can be closed by the delegate on Burn.
+    // token can be closed by the delegate on Burn. We assign CloseAuthority to
+    // the master edition PDA so we can close it on Transfer and revoke it in Revoke.
     if matches!(role, TokenDelegateRole::Utility) {
```

Comment:
I thought about this but it seems like it might be unexpected behavior from a user's perspective. A normal SPL token delegate does not get close authority so it might be surprising if all Token Metadata delegate abstractions did. It would mean that users cannot close their token accounts themselves and might break integrations with marketplaces which use delegates if they are automatically closing and opening token accounts for their users.